docs/src/main/asciidoc/amazon-lambda-http-guide.adoc

```diff
@@ -35,4 +35,110 @@ Quarkus project, you'll need to configure your application for deployment to Ama
+to begin using Quarkus.  One step remains, however.  Your template file should look something like
+https://github.com/awslabs/serverless-application-model/blob/master/examples/apps/hello-world/template.yaml[this].
+
+=== Updating the SAM template file
+
+In an existing AWS project, there will likely already be a YAML file for the SAM CLI to use when developing locally and
+deploying.  However others will need an initial template to work with.  To create that initial template, the following
+plugin can be run from the root of the project:
+
+[source,shell]
+----
+mvn io.quarkus:quarkus-amazon-lambda-http-maven:configure-aws-lambda
+----
+
+If there is no existing template file one will be created with the default name of `template.yaml` and the resource name
+will default to `Quarkus`.  These names might not be preferable or there might be an existing template file to work with.
+In this case, there are two properties that can configure these values:
+
+|===
+|Name |Description
+
+|amazon-lambda-http.template| The template file to use
+|amazon-lambda-http.resource| The resource to manage
+|===
+
+NOTE: Running this plugin in an existing project will attempt to update the existing configuration.  If the template
+file has a different name, the plugin will assume there is no configuration and will generate a new file.
+
+This plugin will update the `Handler` and `Runtime` values for the named resource.  Running this plugin once might be
+sufficient.  However, you might choose to always run this plugin to make sure your template file is always correct.
+e.g., should implementation details on the Quarkus side ever change, it might be nice to have things automatically update
+without having to worry about tracking changes.  The pom can be configured to run automatically by adding the following
+to the `plugins` section:
+
+[source,xml]
+----
+<plugin>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-amazon-lambda-http-maven</artifactId>
+    <version>${quarkus.version}</version>
+    <executions>
+        <execution>
+            <id>amazon-lambda-http</id>
+            <goals>
+                <goal>configure-aws-lambda</goal>
+            </goals>
+            <phase>compile</phase>
+        </execution>
+    </executions>
+</plugin>
+----
+
+This will bind the plugin to the `compile` phase and will ensure that the configuration is always up to date.  If this
+option is chosen some extra configuration will be required since always having to pass the properties above is not ideal.
+It is possible to store these values in `src/main/properties/application.properties` alongside all of the other Quarkus
+configuration values.  In this case, you'll need to define the following entries:
+
+*  quarkus.amazon-lambda-http.resource
+*  quarkus.amazon-lambda-http.template
+
+With these defined, the properties no longer need to be defined when invoking maven.
```

Comment:
`when invoking maven` was a little vague for me and I initially didn't know which command it was referring to. Do it make sense to be a little more explicit?

Reply:
reworded slightly.  see what you think.

Reply:
Looks good!